### PR TITLE
Integrate Manifester with API activation key test

### PR DESCRIPTION
**Pull request by synkd**
The API activation key test module currently contains a single test that
uses a cloned manifest. This PR switches that test to use Manifester to
obtain a manifest (via the session_entitlement_manifest fixture) as a
proof of concept for Robottelo/Manifester integration.

### CODE REVIEW

**Review by JacobCallahan — APPROVED** [review]

**Comment by synkd** [comment]
The above test run failed because the GitLab MR to add conf/manifest.yaml has not been merged yet. So there's no conf/manifest.yaml available in CI yet. I will rerun once that MR is merged.

**Comment by synkd** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_activationkey.py::test_positive_fetch_product_content

**Review by latran — COMMENTED** [review]
PRT still failing.  Please recheck.

**Comment by omkarkhatavkar** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_activationkey.py::test_positive_fetch_product_content

**Comment by synkd** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_activationkey.py::test_positive_fetch_product_content

**Comment by synkd** [comment]
Trying again with https://github.com/SatelliteQE/robottelo/pull/10027 merged. This should get logging working, so I'll have a better idea of where the failure occurs. My current suspicion is that generating the access token is failing and thus the subsequent requests to the RHSM API are unable to authenticate.

**Comment by synkd** [comment]
Here's the relevant log snippet:
```
[D 220907 18:02:13 manifester:43] Generating access token
[D 220907 18:02:13 helpers:13] Sending request to endpoint ['https://sso.redhat.com/auth/realms/redhat-external/protocol/openid-connect/token']
[D 220907 18:02:13 helpers:15] Response status code is 200
[D 220907 18:02:13 helpers:13] Sending request to endpoint ['https://api.access.redhat.com/management/v1/allocations']
[D 220907 18:02:14 helpers:15] Response status code is 400
[D 220907 18:02:14 manifester:43] Generating access token
[D 220907 18:02:14 helpers:13] Sending request to endpoint ['https://sso.redhat.com/auth/realms/redhat-external/protocol/openid-connect/token']
[D 220907 18:02:14 helpers:15] Response status code is 200
```
Manifester is getting a 200 response when generating the access token, then a 400 when trying to access the RHSM API, then it tries to generate a new access token, which it should only do if there is no access token. This suggests that an access token is not being returned initially, and subsequently, the RHSM API is giving a bad request response because an access token is not included in the request. I'm going to trigger another run and try to do some interactive troubleshooting while it's running.

**Comment by synkd** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_activationkey.py::test_positive_fetch_product_content

**Comment by synkd** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_activationkey.py::test_positive_fetch_product_content

**Comment by synkd** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_activationkey.py::test_positive_fetch_product_content

**Comment by synkd** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_activationkey.py::test_positive_fetch_product_content

**Comment by synkd** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_activationkey.py::test_positive_fetch_product_content

**Comment by synkd** [comment]
I have identified the source of the error by debugging on a running Robottelo pod:
```
(Epdb) manifest.allocation
{'error': {'code': 400, 'message': 'invalid version'}}
(Epdb) manifest.sat_version
'sat-6.12.0'
```
The issue is that the value of the `sat_version` parameter passed with the API call to create a subscription allocation must be in the form of `sat-x.y` and that versions are not valid until they have been released. So `sat-6.12` will not be a valid value until the Satellite 6.12 GA release. I'll need to determine how best to clean up this parameter. Once I implement that change, this test should pass.

**Comment by synkd** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_activationkey.py::test_positive_fetch_product_content

File: robottelo/config/__init__.py [hunk 1]
```diff
@@ -173,4 +173,4 @@ def configure_airgun():
     )
 
 
-configure_airgun()
+configure_airgun()
```
Comment (JacobCallahan) [inline robottelo/config/__init__.py:176]: check missing empty line after this

**Review by JacobCallahan — CHANGES_REQUESTED** [review]
reorder imports checks failing. try running pre-commit against all files locally

**Comment by mshriver** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_activationkey.py::test_positive_fetch_product_content

**Comment by devendra104** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_activationkey.py::test_positive_fetch_product_content

**Review by JacobCallahan — APPROVED** [review]

**Review by ogajduse — APPROVED** [review]
ACK based on robottelo-pr-testing/1200 job.

**Review by Gauravtalreja1 — APPROVED** [review]

**Pull request merged**